### PR TITLE
Feat/add mongo start time listener

### PR DESCRIPTION
**Pull request by nettashviro**

### CODE REVIEW

File: src/changeStreamModel.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import mongoose, { Schema } from "mongoose";
+
+const changeStreamSchema = new Schema({
+  eventId: { type: String, unique: true },
+  timeStamp: Number,
+  description: Object,
+  createdAt: { type: Date, default: Date.now, index: { expireAfterSeconds: 3600 } },
```
Comment (Shahar-Y) [inline src/changeStreamModel.ts]: The `3600` should be from config.

File: src/changeStreamModel.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import mongoose, { Schema } from "mongoose";
```
Comment (Shahar-Y) [inline src/changeStreamModel.ts]: Add a documentation to the schema. It's not clear

File: src/index.ts [hunk 3]
```diff
@@ -111,84 +110,108 @@ async function initRabbitConn(rabbituri: string, options: MTROptions) {
...
-    }).on('error', async(err) => {
-        console.log(`error in mongo`);
-        console.log(err);
-        mongoConnection(mongoData, rabbitData, options);
-    });
+    changeStream
+        .on('change', (event: ChangeEvent) => {
+            log(`got mongo change event:  ${event.operationType} in collection:${mongoData.collectionName}`, options);
+
+            // Update event stream d
```
Comment (Shahar-Y) [inline src/index.ts]: ?

File: src/index.ts [hunk 4]
```diff
@@ -111,84 +110,108 @@ async function initRabbitConn(rabbituri: string, options: MTROptions) {
...
-    }));
+        })
+    );
 }
 
+/**
+ * Initiate stream from last event
+ */
+async function initiateChangeStreamStartTime(collectionName: string) {
+    const latestEvent = await changeStreamTrackerModel(collectionName).findOne({}).sort({ timeStamp: -1 });
```
Comment (Shahar-Y) [inline src/index.ts]: Not  `collectionName-events`?

File: src/index.ts [hunk 5]
```diff
@@ -111,84 +110,108 @@ async function initRabbitConn(rabbituri: string, options: MTROptions) {
...
-        console.log(`error in mongo`);
-        console.log(err);
-        mongoConnection(mongoData, rabbitData, options);
-    });
+    changeStream
+        .on('change', (event: ChangeEvent) => {
+            log(`got mongo change event:  ${event.operationType} in collection:${mongoData.collectionName}`, options);
+
+            // Update event stream d
+            changeStreamTrackerModel(mongoData.collectionName).create(
```
Comment (Shahar-Y) [inline src/index.ts]: Where is the object removed?

**Review by Shahar-Y — CHANGES_REQUESTED** [review]
Looks like a good concept of implementation. Notice the object is never deleted (as far as I can see) - Why not use `upsert`?

File: src/index.ts (hunk 3 above)
Comment (nettashviro) [reply src/index.ts]: what?

File: src/index.ts (hunk 5 above)
Comment (nettashviro) [reply src/index.ts]: when it's expired

File: src/index.ts (hunk 4 above)
Comment (nettashviro) [reply src/index.ts]: nope.

File: src/changeStreamModel.ts (hunk 2 above)
Comment (nettashviro) [reply src/changeStreamModel.ts]: done

File: src/index.ts (hunk 3 above)
Comment (Shahar-Y) [reply src/index.ts]: The documentation is not complete

File: src/index.ts (hunk 5 above)
Comment (Shahar-Y) [reply src/index.ts]: How does it make sure that it is not expired by the time the service is up?

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

**Pull request closed without merging**